### PR TITLE
Margin support

### PR DESCRIPTION
**Pull request by Tenchi2xh**
Adds feature for having margins on the left (and on the right if wrapping is enabled) of the body of the presentation (borders at the top and bottom are still full-width):

```markdown
---
patat:
    wrap: true
    margin: 12
...
```

The configuration above renders to:

screen shot 2018-08-25 at 09 51 54

([Comparison with no margin](https://user-images.githubusercontent.com/4116708/44616482-0d138300-a851-11e8-954c-fff648da779b.png))

(I didn't know you were also the author of Lorem Markdownum haha)

As said in #48, I'm not a Haskeller so the code isn't perfect! There may be a better way to have the margin value other than passing it along to multiple functions?

### CODE REVIEW

File: src/Patat/Presentation/Display.hs [hunk 1]
```diff
@@ -241,6 +244,15 @@ prettyBlocks :: Theme -> [Pandoc.Block] -> PP.Doc
 prettyBlocks theme = PP.vcat . map (prettyBlock theme) . filter isVisibleBlock
 
 
+--------------------------------------------------------------------------------
+prettyBlocks' :: Theme -> Int -> [Pandoc.Block] -> PP.Doc
+prettyBlocks' theme margin =
+    let
+        spaces = mconcat (replicate margin PP.space)
+        indent = PP.indent (PP.NotTrimmable spaces) (PP.NotTrimmable spaces) in
+    PP.vcat . map indent . map (prettyBlock theme) . filter isVisibleBlock
```
Comment (jaspervdj) [inline src/Patat/Presentation/Display.hs]: Can this be moved to `withWrapSettings`?  I am not completely sure but I think something like `PP.wrapAt (...) . PP.indent (...margin..) (..margin..)` would do the trick, and that way we only need to pass the margin to `withWrapSettings`.

File: test.sh [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+
+function restore() {
+    mv $PATAT_CONF_BACKUP $PATAT_CONF
+}
+
+if [[ -f "$PATAT_CONF" ]]; then
+    mv $PATAT_CONF $PATAT_CONF_BACKUP
+    trap restore EXIT
+fi
+
```
[the remark this answers is not in the document]
Comment (Tenchi2xh) [reply test.sh]: Neat trick!

File: src/Patat/Presentation/Internal.hs [hunk 3]
```diff
@@ -55,6 +55,7 @@ data Presentation = Presentation
 data PresentationSettings = PresentationSettings
     { psRows             :: !(Maybe (A.FlexibleNum Int))
     , psColumns          :: !(Maybe (A.FlexibleNum Int))
+    , psMargin           :: !(Maybe (A.FlexibleNum Int))
```
[the remark this answers is not in the document]
Comment (Tenchi2xh) [reply src/Patat/Presentation/Internal.hs]: Yeah good idea

File: src/Patat/Presentation/Display.hs (hunk 1 above)
Comment (Tenchi2xh) [reply src/Patat/Presentation/Display.hs]: Didn't have time to take a look at this part yet, I'll see later

**Comment by Tenchi2xh** [comment]
I will have one last feature to PR after this one 🙂 

screen shot 2018-08-26 at 11 26 39

Choosing an `onColor` for the borders applies it to the whole border, if you like it I'll make a PR after this one is finished 😁

**Comment by Tenchi2xh** [comment]
Looks much cleaner now! (Sorry for the commits all moving to the end, I did a rebase on master)

The indenting step is now done in `withWrapSettings`, which I also renamed to `formatWith`: I thought this reads better:

```haskell
PP.putDoc $ formatWith settings $ f canvasSize theme
```

If you'd like another name let me know

**Comment by jaspervdj** [comment]
Thanks!

I'm happy to take another PR for the border colour.

**Pull request merged**